File: coding_practice/Lua/fizzbuzz/fizzbuzz.lua
#!/usr/bin/lua

for i = 1, 100, 1 do
    msg = ""
    if i % 3 == 0 then
        msg = msg .. "fizz"
    end
    if i % 5 == 0 then
        msg = msg .. "buzz"
    end
    if string.len(msg) == 0 then
        msg = msg .. i
    end
    print(msg)
end

File: coding-practice/Perl/fizzbuzz/fizzbuzz.pl
#!/usr/bin/perl -w

package Main;

use strict;
use warnings;
use FindBin;
use Cwd;

my $exec_name = "$FindBin::RealScript";
my $exec_path = "$FindBin::RealBin/$FindBin::RealScript";
my $exec_dir = "$FindBin::RealBin";
my $original_cwd = getcwd();

sub main {
    my $exit_code = 0;
    my $str;
    for (my $i = 1; $i <= 100; $i++) {
        $str = "";
        if ($i % 3 == 0) {
            $str .= "fizz";
        }
        if ($i % 5 == 0) {
            $str .= "buzz";
        }
        if (length($str) == 0) {
            $str .= $i;
        }
        STDOUT->printflush("$str\n");
    }
    return $exit_code;
}

1;

exit(Main::main());
